libs/aws/langchain_aws/tools/browser_session_manager.py

```diff
@@ -0,0 +1,259 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Dict, Tuple
+
+if TYPE_CHECKING:
+    from playwright.async_api import Browser as AsyncBrowser
+    from playwright.sync_api import Browser as SyncBrowser
+
+from bedrock_agentcore.tools.browser_client import BrowserClient
+
+logger = logging.getLogger(__name__)
+
+
+class BrowserSessionManager:
+    """
+    Manages browser sessions for different threads.
+
+    This class maintains separate browser sessions for different threads,
+    enabling concurrent usage of browsers in multi-threaded environments.
+    Browsers are created lazily only when needed by tools.
+    """
+
+    def __init__(self, region: str = "us-west-2"):
+        """
+        Initialize the browser session manager.
+
+        Args:
+            region: AWS region for browser client
+        """
+        self.region = region
+        self._async_sessions: Dict[str, Tuple[BrowserClient, AsyncBrowser]] = {}
+        self._sync_sessions: Dict[str, Tuple[BrowserClient, SyncBrowser]] = {}
+
+    async def get_async_browser(self, thread_id: str) -> AsyncBrowser:
+        """
+        Get or create an async browser for the specified thread.
+
+        Args:
+            thread_id: Unique identifier for the thread requesting the browser
+
+        Returns:
+            An async browser instance specific to the thread
+        """
+        if thread_id in self._async_sessions:
+            return self._async_sessions[thread_id][1]
```

Comment:
I assume a browser session can't be used by two different callers at the same time? Worth adding some note or some protection for it?

Reply:
Thanks @nfcampos, I'm looking into adding thread protection - will update the PR by early tomorrow